libr/util/str.c

```diff
@@ -217,26 +219,44 @@ R_API int r_str_binstr2bin(const char *str, ut8 *out, int outlen) {
 				if (str[j] == '1') {
 					ret |= (1 << k);
 				} else if (str[j] != '0') {
-					return n;
+					return r_strbuf_drain (buf);
 				}
 			}
 		}
-		out[n++] = ret;
-		if (n == outlen) {
-			return n;
-		}
+		r_strbuf_appendf (buf, "%c", ret);
 	}
-	return n;
+
+	return r_strbuf_drain (buf);
+}
+
+R_API char *r_str_str2binstr(const char *str) {
```

Comment:
i think the input for this should be (const ut8*, size_t len), because the output should handle null bytes too. the same thing may work well with strings if needed. that's why i was suggesting to use the `r_mem_tobinstring()` instead. but the implementation looks good 👍